-1

Tested at: bf2b38677f5a6131186b7dc76a300b01e26dfabe

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18041/18844/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-31-2300/src/RecoTracker/FinalTrackSelectors/src/TrackAlgoPriorityOrder.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-31-2300/src/RecoTracker/FinalTrackSelectors/src/TrackCollectionCloner.cc 
>> Building binary trackAlgoPriorityOrder_t
Copying tmp/slc6_amd64_gcc530/src/RecoTracker/FinalTrackSelectors/test/trackAlgoPriorityOrder_t/trackAlgoPriorityOrder_t to productstore area:
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-31-2300/src/RecoTracker/FinalTrackSelectors/src/TrackMVAClassifierBase.cc: In member function 'virtual void TrackMVAClassifierBase::produce(edm::Event&, const edm::EventSetup&)':
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-31-2300/src/RecoTracker/FinalTrackSelectors/src/TrackMVAClassifierBase.cc:76:7: error: 'LogWarning' is not a member of 'edm'
       edm::LogWarning("TrackMVAClassifierBase") << "ignoreVertices is set to False in the configuration, but the vertex collection is not valid"; 
       ^
gmake: *** [tmp/slc6_amd64_gcc530/src/RecoTracker/FinalTrackSelectors/src/RecoTrackerFinalTrackSelectors/TrackMVAClassifierBase.o] Error 1
>> Building shared library tmp/slc6_amd64_gcc530/src/RecoTracker/FinalTrackSelectors/src/RecoTrackerFinalTrackSelectors/libRecoTrackerFinalTrackSelectors.so
c++: error: tmp/slc6_amd64_gcc530/src/RecoTracker/FinalTrackSelectors/src/RecoTrackerFinalTrackSelectors/TrackMVAClassifierBase.o: No such file or directory
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 48 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-31-2300/src/RecoTracker/FinalTrackSelectors/src/TrackCollectionCloner.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-31-2300/src/RecoTracker/FinalTrackSelectors/src/ES_TrackAlgoPriorityOrder.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-31-2300/src/RecoTracker/FinalTrackSelectors/plugins/CosmicTrackSplitter.cc:408:27: warning: expression with side effects will be evaluated despite being used as an operand to 'typeid' [-Wpotentially-evaluated-expression]
                        ", type  = " << typeid(*hitR.first).name();
                                               ^
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_1_X_2017-03-31-2300/src/RecoTracker/FinalTrackSelectors/src/TrackMVAClassifierBase.cc:76:12: error: no member named 'LogWarning' in namespace 'edm'
      edm::LogWarning("TrackMVAClassifierBase") << "ignoreVertices is set to False in the configuration, but the vertex collection is not valid"; 
      ~~~~~^
1 warning generated.
1 error generated.
gmake: *** [tmp/slc6_amd64_gcc530/src/RecoTracker/FinalTrackSelectors/src/RecoTrackerFinalTrackSelectors/TrackMVAClassifierBase.o] Error 1
</pre>

--- a/RecoTracker/FinalTrackSelectors/src/TrackMVAClassifierBase.cc
+++ b/RecoTracker/FinalTrackSelectors/src/TrackMVAClassifierBase.cc
@@ -3,6 +3,7 @@
 #include "CondFormats/DataRecord/interface/GBRWrapperRcd.h"
 #include "FWCore/Framework/interface/EventSetup.h"
 #include "FWCore/Framework/interface/ESHandle.h" 
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 #include "DataFormats/TrackReco/interface/Track.h"
 #include "DataFormats/VertexReco/interface/Vertex.h"

